hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetCommandLine.java

```diff
@@ -712,4 +814,373 @@ static Throwable peel(Throwable e) {
+                            // End the block; arg could be empty, but that's fine
+                            multiLineCommentStart = -1;
+                        }
+                    } else if (oneLineCommentStart == -1 && isOneLineComment(line, i)) {
+                        oneLineCommentStart = i;
+                    } else if (oneLineCommentStart >= 0) {
+                        if (currentChar == '\n') {
+                            // End the block; arg could be empty, but that's fine
+                            oneLineCommentStart = -1;
+                        }
+                    } else {
+                        // Not in a quote or comment block
+                        containsNonCommentData = true;
+                        if (!Character.isWhitespace(currentChar)) {
+                            lastNonQuoteCommentIndex = i;
+                        }
+                    }
+                }
+            }
+
+            if (isEofOnEscapedNewLine() && isEscapeChar(line, line.length() - 1)) {
+                throw new EOFError(-1, cursor, "Escaped new line");
+            }
+
+            if (isEofOnUnclosedQuote() && quoteStart >= 0) {
+                throw new EOFError(-1, quoteStart, "Missing closing quote",
+                        line.charAt(quoteStart) == '\'' ? "quote" : "dquote");
+            }
+
+            if (multiLineCommentStart != -1) {
+                throw new EOFError(-1, cursor, "Missing end of comment", "**");
+            }
+
+            final int lastNonQuoteCommentIndex1 =
+                    lastNonQuoteCommentIndex == line.length() - 1
+                            && lastNonQuoteCommentIndex - 1 >= 0
+                            ? lastNonQuoteCommentIndex - 1 : lastNonQuoteCommentIndex;
+            if (containsNonCommentData
+                    && !isLineFinishedWithSemicolon(
+                    lastNonQuoteCommentIndex1, line)) {
+                throw new EOFError(-1, cursor, "Missing semicolon (;)");
+            }
+        }
+
+        /**
+         * Returns whether a line (already trimmed) ends with a semicolon that
+         * is not commented with one line comment.
+         *
+         * <p>ASSUMPTION: to have correct behavior, this method must be
+         * called after quote and multi-line comments check calls, which implies that
+         * there are no non-finished quotations or multi-line comments.
+         *
+         * @param buffer Input line to check for ending with ';'
+         * @return true if the ends with non-commented ';'
+         */
+        private boolean isLineFinishedWithSemicolon(
+                final int lastNonQuoteCommentIndex, final CharSequence buffer) {
+            final String line = buffer.toString();
+            boolean lineEmptyOrFinishedWithSemicolon = line.isEmpty();
+            boolean requiredSemicolon = false;
+            String[] oneLineComments = {"#", "--"};
```

Comment:
`#` is not one line comment in sql. It's probably a feature of sqline.

Reply:
Removed it. Also, I replaced the isOneLineComment()and isMultilineComment() methods with one liners.